QEfficient/lora/auto.py

```diff
@@ -0,0 +1,396 @@
+            for i in range(self.max_num_adapters)
+        ), "Not all adapters have the same ranks"
+        self.lora_rank = list(self.adapter_configs.values())[0].r
+
+        # do the module replacement
+        _, transformed = LoraModelInputsTransform.apply(self.model)
+
+        self.target_modules_for_all_adapters = list(self.adapter_configs.values())[0].target_modules
+        _, transformed = TargetModulesTransform.apply(
+            self.model, self.target_modules_for_all_adapters, self.lora_rank, self.max_num_adapters
+        )
+
+        # load_weight to model
+        self.load_adapter_weights_to_model()
+
+    def export(self, **kwargs) -> str:
+        """
+        Exports the model to ``ONNX`` format using ``torch.onnx.export``.
+        The model should already be transformed i.e. ``self.is_transformed`` should be ``True``.
+        Otherwise, this will raise an ``AssertionError``.
+        We currently don't support exporting non-transformed models. Please refer to the ``convert_to_cloud_bertstyle`` function in the **Low-Level API** for a legacy function that supports this."
+
+        ``Optional`` Args:
+            does not any arguments.
+
+        Raises:
+            :AttributeError: If ``pretrained_model_name_or_path`` is a path, this function needs model card name of the model so that it can distinguish between directories while saving the ``ONNX`` files generated. So, user needs to pass ``model_card_name`` as a valid ``string`` in that case, Otherwise this will raise the error.
+
+        Returns:
+            :str: Path of the generated ``ONNX`` graph.
+        """
+
+        self.full_batch_size = kwargs.get("full_batch_size", self.full_batch_size)
+        export_dir = kwargs.get("export_dir", None)
+
+        # obtain all necessary information to initialize the model
+        self.init_adapter_model()
+
+        assert self.is_transformed, "Please first run transform on the QEFFAutoModelForCausalLM object"
+
+        # Caching export onnx
+        if export_dir is None:
+            model_card_dir = os.path.join(QEFF_MODELS_DIR, str(self.model_card_name))
+            export_dir = Path(model_card_dir).with_name(str(self.model_card_name).split("/")[1] + "-" + self.model_hash)
+        else:
+            export_dir = Path(export_dir).with_name(export_dir.name + "-" + self.model_hash)
+        onnx_dir_path = os.path.join(export_dir, "onnx")
+        model_base_name = self.model_card_name.replace("/", "_") + "_kv"
+        onnx_path = os.path.join(onnx_dir_path, f"{model_base_name}.onnx")
+
+        if Path(onnx_path).is_file():
+            self.onnx_path = onnx_path
+            print(f"Using existing onnx path:-{self.onnx_path}")
+            return self.onnx_path
+
+        # Export
+        os.makedirs(onnx_dir_path, exist_ok=True)
+        _, onnx_model_path = QEfficient.export(
+            model_name=self.model_card_name,
```

Comment:
model_name should be model_base_name

Reply:
This is to align with the first onnx name generated by QEfficient.export(), so will keep it